File: src/_02_StructuralPattern/_02_05_Facade/EmailSettingsFacade.js
export default class EmailSettingsFacade {
  /** @type { string } */
  #host;

  /** @param { string } host */
  constructor(host) {
    this.#host = host;
  }

  // host
  /** @returns { string } */
  getHost() {
    return this.#host;
  }
  /** @param { string } host */
  setHost(host) {
    this.#host = host;
  }
}

File: src/_02_StructuralPattern/_02_05_Facade/java/after/EmailSettingsFacade.java
package _02_StructuralPattern._02_05_Facade.java.after;

public class EmailSettingsFacade {
  
  private String host;

  public EmailSettingsFacade(String host) {
    this.host = host;
  }

  // host
  public String getHost() {
    return host;
  }
  public void setHost(String host) {
    this.host = host;
  }
  
}
